apps/teamsjs-diagnostic-app/api/src/functions/getUserProfile.ts

```diff
@@ -0,0 +1,204 @@
+
+  // Initialize response.
+  const res: HttpResponseInit = {
+    status: 200,
+  };
+  const body = Object();
+
+  // Put an echo into response body.
+  body.receivedHTTPRequestBody = (await req.text()) || "";
+
+  // Prepare access token.
+  const accessToken: string = req.headers.get("Authorization")?.replace("Bearer ", "").trim();
+  if (!accessToken) {
+    return {
+      status: 400,
+      body: JSON.stringify({
+        error: "No access token was found in request header.",
+      }),
+    };
+  }
+
+  const oboAuthConfig: OnBehalfOfCredentialAuthConfig = {
+    authorityHost: config.authorityHost,
+    clientId: config.clientId,
+    tenantId: config.tenantId,
+    clientSecret: config.clientSecret,
+  };
+
+  let oboCredential: OnBehalfOfUserCredential;
+  try {
+    oboCredential = new OnBehalfOfUserCredential(accessToken, oboAuthConfig);
+  } catch (e) {
+    context.error(e);
+    return {
+      status: 500,
+      body: JSON.stringify({
+        error:
+          "Failed to construct OnBehalfOfUserCredential using your accessToken. " +
+          "Ensure your function app is configured with the right Microsoft Entra App registration.",
+      }),
+    };
+  }
+
+  // Query user's information from the access token.
+  try {
+    const currentUser: UserInfo = await oboCredential.getUserInfo();
+    if (currentUser && currentUser.displayName) {
+      body.userInfoMessage = `User display name is ${currentUser.displayName}.`;
+    } else {
+      body.userInfoMessage = "No user information was found in access token.";
+    }
+  } catch (e) {
+    context.error(e);
+    return {
+      status: 400,
+      body: JSON.stringify({
+        error: "Access token is invalid.",
+      }),
+    };
+  }
+
+  // Create a graph client with default scope to access user's Microsoft 365 data after user has consented.
+  try {
+    // Create an instance of the TokenCredentialAuthenticationProvider by passing the tokenCredential instance and options to the constructor
+    const authProvider = new TokenCredentialAuthenticationProvider(oboCredential, {
+      scopes: ["https://graph.microsoft.com/.default"],
+    });
+
+    // Initialize Graph client instance with authProvider
+    const graphClient = Client.initWithMiddleware({
+      authProvider: authProvider,
+    });
+
+    body.graphClientMessage = await graphClient.api("/me").get();
+  } catch (e) {
+    context.error(e);
+    return {
+      status: 500,
+      body: JSON.stringify({
+        error:
+          "Failed to retrieve user profile from Microsoft Graph. The application may not be authorized.",
+      }),
+    };
+  }
+  res.body = JSON.stringify(body);
+
+  return res;
+}
+
+app.http("getUserProfile", {
+  methods: ["GET", "POST"],
+  authLevel: "anonymous",
+  handler: getUserProfile,
+});
+
+// You can replace the codes above from the function body with comment "Query user's information from the access token." to the end
```

Comment:
Do we want to keep the following commented out segments?

Reply:
I haven't modified that file yet because that was part of the Teams Toolkit template tab app